### ISSUE TITLE
SIGSEGV on resourceComponentRead()

### ISSUE THREAD

**Issue by Dombo** [sbecker59/terraform-provider-statuspage#126]
Hey, came across this weird panic - it's intermittent so I'm wondering if it's related to statuspages API response. Subsequent retries got the plan-apply to succeed without error. I'm managing a few hundred components per plan-apply as well, perhaps that could be a contributor.

### Terraform Version
`1.0.0`

### Affected Resource(s)
- statuspage_component_group
- statuspage_component

### Panic Output
Here's a [gist](https://gist.github.com/Dombo/c577d3c48d31081439dfef9e807a9eaf)

### Important Factoids
Terraform was running inside TFC , the same workspace has successfully applied numerous times.

**Comment by sbecker59** [comment]
Hi @Dombo 

First, thank you for your issue.

There is indeed a problem in my code, I log the API response status before the `if err.Error() != "" {` , which poses a problem if I have no return from the API.

```go
component, h, err := statuspageClientV1.ComponentsApi.GetPagesPageIdComponentsComponentId(authV1, d.Get("page_id").(string), d.Id()).Execute()
log.Printf("[INFO] StatusCode %d", h.StatusCode)
if err.Error() != "" {
	return TranslateClientErrorDiag(err, "failed to get component using Status Page API")
}
```

The result would have been finally the same, there would have been the following exception `failed to get component using Status Page API` [View code](https://github.com/sbecker59/terraform-provider-statuspage/blob/main/statuspage/resource_component.go#L25)

You should know that there is a rate limit of the api proposed by [statuspage](https://developer.statuspage.io/#section/Rate-Limiting).

Rate limit - StatuspageAPI
---

Each API token is limited to 1 request / second as measured on a 60 second rolling window. To get this limit increased, please contact us at https://support.atlassian.com/contact

Error codes 420 or 429 indicate that you have exceeded the rate limit and the request has been rejected.

To overcome these problems, I implemented a retry in case of http code at 429 [View code](https://github.com/sbecker59/terraform-provider-statuspage/blob/main/statuspage/internal/go-retryablehttp/client.go#L47). There may be things to improve on this side.

What do you think ?
Do you have the possibility of proposing a pull request in order to improve the solution ?

**Comment by sbecker59** [comment]
The error on the logs has been corrected in release [release v1.0.0](https://github.com/sbecker59/terraform-provider-statuspage/releases/tag/v1.0.0)
